### PR TITLE
[15.0][MIG] sale_stock_available_info_popup: Migration to 15.0

### PR DESCRIPTION
**Pull request by victoralmau**
Migration to 15.0

Please @pedrobaeza and @chienandalu can you review it?

@Tecnativa TT36989

### CODE REVIEW

File: sale_stock_available_info_popup/tests/test_sale_stock_available_info_popup.py [hunk 1]
```diff
@@ -4,19 +4,21 @@
...
     @classmethod
     def setUpClass(cls):
         super().setUpClass()
-        # Scratch some seconds from the tracking stuff that we won't need
-        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
-        cls.env.ref("base.user_demo").groups_id += cls.env.ref("stock.group_stock_user")
+        cls.stock_user = new_test_user(
+            cls.env,
+            login="stock_user",
+            groups="sales_team.group_sale_salesman,stock.group_stock_user",
```
Comment (pedrobaeza) [inline sale_stock_available_info_popup/tests/test_sale_stock_available_info_popup.py]: The idea is to not have stock permissions to see that information.

File: sale_stock_available_info_popup/tests/test_sale_stock_available_info_popup.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        sale_form.partner_id = self.partner
+        sale_form.user_id = self.stock_user
+        return sale_form
+
+    def _add_sale_line(self, sale_form, product, qty=1):
+        with sale_form.order_line.new() as line:
+            line.product_id = self.product
+            line.product_uom_qty = qty
+
+    @users("stock_user")
```
Comment (pedrobaeza) [inline sale_stock_available_info_popup/tests/test_sale_stock_available_info_popup.py]: Don't remove `admin` from both tests. We want to know if it fails due to permissions (the other user) or feature (the admin user).

File: sale_stock_available_info_popup/tests/test_sale_stock_available_info_popup.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    def test_immediately_usable_qty_today(self):
+        sale_form = self._create_sale()
+        self._add_sale_line(sale_form, self.product)
+        sale = sale_form.save()
+        self.assertAlmostEqual(
+            sale.order_line.immediately_usable_qty_today,
+            self.product.immediately_usable_qty,
```
Comment (pedrobaeza) [inline sale_stock_available_info_popup/tests/test_sale_stock_available_info_popup.py]: Can you use an absolute value?

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
I don't see the code for changing the color of the icon according the new variable.

File: sale_stock_available_info_popup/models/sale_order.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        self.immediately_usable_qty_today = False
+        for line in self.sorted(key=lambda r: r.sequence):
+            if line.order_id.commitment_date:
+                date = line.order_id.commitment_date
+            else:
+                date = line._expected_date()
+            if not line.display_qty_widget:
+                continue
+            product = line.product_id.with_context(
+                to_date=date, warehouse=line.warehouse_id.id
```
Comment (chienandalu) [inline sale_stock_available_info_popup/models/sale_order.py]: Using `line.scheduled_date` should do the same:
```suggestion
            if not line.display_qty_widget:
                continue
            product = line.product_id.with_context(
                to_date=line.scheduled_date, warehouse=line.warehouse_id.id
```

**Comment by chienandalu** [comment]
BTW. You put the repos's name in the PRs title instead of the module's :sweat_smile:

File: sale_stock_available_info_popup/static/src/xml/qty_at_date.xml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+            t-jquery="t[t-if^='widget.data.virtual_available_at_date']"
+            t-operation="attributes"
+        >
+            <attribute
+                name="t-if"
+            >widget.data.immediately_usable_qty_today &lt; widget.data.qty_to_deliver and !widget.data.is_mto</attribute>
+        </t>
+    </div>
+    <div t-extend="sale_stock.QtyDetailPopOver">
+        <t t-jquery="t[t-if*='!data.is_mto']" t-operation="append">
```
Comment (chienandalu) [inline sale_stock_available_info_popup/static/src/xml/qty_at_date.xml]: The field doesn't show up in the lines popup. I'd say you need to refine this selectors.

**Review by chienandalu — CHANGES_REQUESTED** [review]

File: stock_demand_estimate/tests/test_stock_demand_estimate.py [hunk 6]
```diff
@@ -3,10 +3,10 @@
 
 from datetime import date, timedelta as td
 
-from odoo.tests.common import SavepointCase
+from odoo.tests.common import TransactionCase
```
Comment (rousseldenis) [inline stock_demand_estimate/tests/test_stock_demand_estimate.py]: @victoralmau Could you put this in another PR as you don't depend on that module here ?

File: sale_stock_available_info_popup/static/src/xml/qty_at_date.xml (hunk 5 above)
Comment (victoralmau) [reply sale_stock_available_info_popup/static/src/xml/qty_at_date.xml:16]: It seems that v14 is also incorrect.

File: stock_demand_estimate/tests/test_stock_demand_estimate.py (hunk 6 above)
Comment (victoralmau) [reply stock_demand_estimate/tests/test_stock_demand_estimate.py]: https://github.com/OCA/stock-logistics-warehouse/pull/1530

File: sale_stock_available_info_popup/static/src/xml/qty_at_date.xml (hunk 5 above)
Comment (chienandalu) [reply sale_stock_available_info_popup/static/src/xml/qty_at_date.xml]: The module isn't migrated to v14 or is it? :thinking:

Comment (victoralmau) [reply sale_stock_available_info_popup/static/src/xml/qty_at_date.xml]: Sorry, it is not migrated to v14. Changes done (it´s already solved).

**Review by chienandalu — APPROVED** [review]
Thanks! :+1:

**Review by pedrobaeza — APPROVED** [review]
